packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol

```diff
@@ -302,44 +302,51 @@ contract OVM_L1CrossDomainMessenger is
         address _sender,
         bytes memory _message,
         uint256 _queueIndex,
-        uint32 _gasLimit
+        uint32 _oldGasLimit,
+        uint32 _newGasLimit
     )
         override
         public
     {
-        // Verify that the message is in the queue:
-        address canonicalTransactionChain = resolve("OVM_CanonicalTransactionChain");
-        Lib_OVMCodec.QueueElement memory element =
-            iOVM_CanonicalTransactionChain(canonicalTransactionChain).getQueueElement(_queueIndex);
+        // Compute the calldata that was originally used to send the message.
```

Comment:
Maybe its worth requiring that the new gas limit is larger than the old gas limit?

Reply:
Although I'm inclined to agree, I think it's best not to be opinionated about it. I bet someone will eventually have some reason to replay the transaction with a lower gas limit.